### PR TITLE
Allow Session expiry to be configured

### PR DESCRIPTION
**Pull request by hardillb**
part of #4201

## Description

Allows the session expiry to be configured using

```
sessions:
  maxDuration: <time in seconds>
  maxIdleDuration: <time in seconds>
```

Defaults are 7 days for `maxDuration` and 32 hours for `maxIdleDuration`. If `maxIdleDuration` is more than `maxDuration` Forge app will fail to start with an error.

## Related Issue(s)

#4201

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts
 - [x] Changes `flowforge.yml`?
    - [x] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template https://github.com/FlowFuse/helm/pull/443
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

No need to raise CloudProject issue as not changing on FF Cloud

### CODE REVIEW

File: forge/routes/auth/index.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const fp = require('fastify-plugin')
 // This defines how long the session cookie is valid for. This should match
 // the max session age defined in `forge/db/controllers/Session.DEFAULT_WEB_SESSION_EXPIRY
 // albeit in secs not millisecs due to cookie maxAge requirements
+// this can be overridden by `sessions.maxDurartion`
```
Comment (knolleary) [inline forge/routes/auth/index.js:25]:
```suggestion
// this can be overridden by `sessions.maxDuration` in config yml file
```

**Review by knolleary — APPROVED** [review]
One typo. Was in two minds over throwing an error if idle > max, rather than just capping the value. But happy to go with this.

**Review by Steve-Mcl — APPROVED** [review]
Approved (with 1 v minor optional suggestion)

File: forge/config/index.js [hunk 2]
```diff
@@ -110,6 +110,20 @@ module.exports = {
             }
         }
 
+        // need to check that maxIdleDuration is less than maxDuration
+        if (config.sessions) {
+            if (config.sessions.maxIdleDuration && config.sessions.maxDuration) {
+                if (config.sessions.maxIdleDuration > config.sessions.maxDuration) {
+                    throw new Error('Session maxIdleDuration longer than maxDuration')
+                    // config.sessions.maxIdleDuration = config.sessions.maxDuration
```
[the remark this answers is not in the document]
Comment (hardillb) [reply forge/config/index.js:118]: Left as alternative, see Nick's comment about being unsure if we should exit or just change the value.

**Pull request merged**